k9mail/src/test/java/com/fsck/k9/helper/MergeCursorTest.java

```diff
@@ -0,0 +1,450 @@
+package com.fsck.k9.helper;
+
+import java.util.Comparator;
+
+import android.content.ContentResolver;
+import android.database.CharArrayBuffer;
+import android.database.ContentObserver;
+import android.database.Cursor;
+import android.database.DataSetObserver;
+import android.net.Uri;
+
+import com.fsck.k9.K9RobolectricTestRunner;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.robolectric.annotation.Config;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(K9RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class MergeCursorTest {
+
+    @Mock
+    Cursor mCursor1;
+    @Mock
+    Cursor mCursor2;
+
+    private Cursor[] mCursors;
+
+    private Comparator<Cursor> mComparator = new Comparator<Cursor>() {
+
+        @Override
+        public int compare(Cursor cursor1, Cursor cursor2) {
+            int o1 = cursor1.getInt(0);
+            int o2 = cursor2.getInt(0);
+
+            return o1 - o2;
+        }
+
+    };
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        mCursors = new Cursor[2];
+
+        mCursors[0] = mCursor1;
+        mCursors[1] = mCursor2;
+    }
+
+    @Test
+    public void testCursorAreResetOnMergeCursorCreation() {
+        reset(mCursor1);
+        reset(mCursor2);
+        when(mCursor1.moveToPosition(-1)).thenAnswer(new Answer<Boolean>() {
+            @Override
+            public Boolean answer(InvocationOnMock invocation) throws Throwable {
+                when(mCursor1.getPosition()).thenReturn(-1);
+                return true;
+            }
+        });
+        when(mCursor2.moveToPosition(-1)).thenAnswer(new Answer<Boolean>() {
+            @Override
+            public Boolean answer(InvocationOnMock invocation) throws Throwable {
+                when(mCursor2.getPosition()).thenReturn(-1);
+                return true;
+            }
+        });
+
+        new MergeCursor(mCursors, mComparator);
+
+        assertEquals(-1, mCursor1.getPosition());
+        assertEquals(-1, mCursor2.getPosition());
+    }
+
+    @Test
+    public void testClose() {
+        reset(mCursor1);
+        reset(mCursor2);
+
+        MergeCursor mergeCursor = new MergeCursor(mCursors, mComparator);
+        mergeCursor.close();
+
+        verify(mCursor1).close();
+        verify(mCursor2).close();
+    }
+
+    @Test
+    public void testCopyStringToBuffer() {
```

Comment:
There's not much point in testing methods that simply call through to `mActiveCursor`. They're trivial.

What would be useful are tests that check if `mActiveCursor` is advanced correctly when moving to the next position etc.